homeassistant/components/climate/evohome.py

```diff
@@ -53,156 +65,420 @@
     EVO_CUSTOM: STATE_AUTO,
     EVO_HEATOFF: STATE_OFF
 }
-
-HA_STATE_TO_EVO = {
+HA_STATE_TO_TCS = {
     STATE_AUTO: EVO_AUTO,
     STATE_ECO: EVO_AUTOECO,
     STATE_OFF: EVO_HEATOFF
 }
+TCS_OP_LIST = list(HA_STATE_TO_TCS)
+
+# the Zones' opmode; their state is usually 'inherited' from the TCS
+EVO_FOLLOW = 'FollowSchedule'
+EVO_TEMPOVER = 'TemporaryOverride'
+EVO_PERMOVER = 'PermanentOverride'
+
+# for the Zones...
+ZONE_STATE_TO_HA = {
+    EVO_FOLLOW: STATE_AUTO,
+    EVO_TEMPOVER: STATE_MANUAL,
+    EVO_PERMOVER: STATE_MANUAL
+}
+HA_STATE_TO_ZONE = {
+    STATE_AUTO: EVO_FOLLOW,
+    STATE_MANUAL: EVO_PERMOVER
+}
+ZONE_OP_LIST = list(HA_STATE_TO_ZONE)
 
-HA_OP_LIST = list(HA_STATE_TO_EVO)
-
-# these are used to help prevent E501 (line too long) violations
-GWS = 'gateways'
-TCS = 'temperatureControlSystems'
-
-# debug codes - these happen occasionally, but the cause is unknown
-EVO_DEBUG_NO_RECENT_UPDATES = '0x01'
-EVO_DEBUG_NO_STATUS = '0x02'
-
-
-def setup_platform(hass, config, add_entities, discovery_info=None):
-    """Create a Honeywell (EMEA/EU) evohome CH/DHW system.
-
-    An evohome system consists of: a controller, with 0-12 heating zones (e.g.
-    TRVs, relays) and, optionally, a DHW controller (a HW boiler).
 
-    Here, we add the controller only.
-    """
+async def async_setup_platform(hass, hass_config, async_add_entities,
+                               discovery_info=None):
+    """Create the evohome Controller, and its Zones, if any."""
     evo_data = hass.data[DATA_EVOHOME]
 
     client = evo_data['client']
     loc_idx = evo_data['params'][CONF_LOCATION_IDX]
 
-    # evohomeclient has no defined way of accessing non-default location other
-    # than using a protected member, such as below
+    # evohomeclient has exposed no means of accessing non-default location
+    # (i.e. loc_idx > 0) other than using a protected member, such as below
     tcs_obj_ref = client.locations[loc_idx]._gateways[0]._control_systems[0]    # noqa E501; pylint: disable=protected-access
 
     _LOGGER.debug(
-        "setup_platform(): Found Controller: id: %s [%s], type: %s",
-        tcs_obj_ref.systemId,
+        "setup_platform(): Found Controller, id=%s, name=%s (location_idx=%s)",
+        tcs_obj_ref.systemId + " [" + tcs_obj_ref.modelType + "]",
```

Comment:
Keep all the strings part of the logging message, and pass in the arguments using formatting operators.

Reply:
Ha! it was a cheat to fit <80 chars - fixed now